junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestExtension.java

```diff
@@ -98,14 +107,80 @@ public boolean mayReturnZeroTestTemplateInvocationContexts(ExtensionContext exte
 	}
 
 	private ParameterizedTestMethodContext getMethodContext(ExtensionContext extensionContext) {
-		return getStore(extensionContext)//
+		return getStoreInMethodNamespace(extensionContext)//
 				.get(METHOD_CONTEXT_KEY, ParameterizedTestMethodContext.class);
 	}
 
-	private ExtensionContext.Store getStore(ExtensionContext context) {
+	private ExtensionContext.Store getStoreInMethodNamespace(ExtensionContext context) {
 		return context.getStore(Namespace.create(ParameterizedTestExtension.class, context.getRequiredTestMethod()));
 	}
 
+	private ExtensionContext.Store getStoreInExtensionNamespace(ExtensionContext context) {
+		return context.getStore(Namespace.create(ParameterizedTestExtension.class));
```

Comment:
We can store it in the root context since the configuration parameters are the same for all extension contexts. Otherwise, it won't get reused by other parameterized tests because they use a different extension context.

```suggestion
		return context.getRoot().getStore(Namespace.create(ParameterizedTestExtension.class));
```